### PR TITLE
Refactor Site Preview and Block Preview (Admin+Site)

### PR DESCRIPTION
**Pull request by nsams**
This rather large PR consists of smaller commits each trying to refactor parts of the preview systems.

The main goal was to understand (and document) what we have and how things are connected. On the way I did also rename/move stuff to separate block preview and site preview.

This PR should only refactor, not change anything.

### CODE REVIEW

File: packages/admin/cms-admin/src/preview/SitePreview.tsx [hunk 1]
```diff
@@ -21,35 +21,41 @@ interface SiteState {
     includeInvisibleBlocks: boolean;
 }
 
-interface Props extends RouteComponentProps {
+interface Props {
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/preview/SitePreview.tsx]: Is this a breaking change? Won't TS complain about passing too many props to the SitePreview component?

File: packages/site/cms-site/src/blocks/ExternalLinkBlock.tsx [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
     children: React.ReactElement;
 }
 
 export function ExternalLinkBlock({ data: { targetUrl, openInNewWindow }, children }: ExternalLinkBlockProps): React.ReactElement {
-    const iframe = useIFrameBridge();
+    const iframe = useSitePreviewIFrameBridge();
+    const preview = usePreview();
 
-    if (iframe.hasBridge) {
+    if (preview.previewType === "SitePreview" || preview.previewType === "BlockPreview") {
```
Comment (johnnyomair) [inline packages/site/cms-site/src/blocks/ExternalLinkBlock.tsx:16]: Shouldn't the onClick handler only be registered for the "SitePreview" preview type?

File: packages/site/cms-site/src/index.ts [hunk 3]
```diff
@@ -17,9 +17,10 @@ export type { ImageDimensions } from "./image/Image";
...
 export { Link } from "./link/Link";
 export { BlockPreviewProvider } from "./preview/BlockPreviewProvider";
-export { PreviewPage } from "./preview/PreviewPage";
-export { SitePreviewProvider } from "./preview/SitePreviewProvider";
 export { usePreview } from "./preview/usePreview";
 export { parsePreviewState } from "./preview/utils";
 export { PreviewSkeleton } from "./previewskeleton/PreviewSkeleton";
 export { useRouter } from "./router/useRouter";
+export { SitePreviewPage } from "./sitePreview/SitePreviewPage";
+export { SitePreviewPage as PreviewPage } from "./sitePreview/SitePreviewPage"; //legacy
```
Comment (johnnyomair) [inline packages/site/cms-site/src/index.ts:26]: We should add JSDoc `@deprecated` for better IDE integration

File: packages/admin/cms-admin/src/preview/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
+Comet has two different previews:
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/preview/README.md:1]: Nice 👍🏼  Maybe we can re-use parts of this documentation of https://docs.comet-dxp.com

File: packages/site/cms-site/src/index.ts [hunk 5]
```diff
@@ -18,7 +18,8 @@ export { calculateInheritAspectRatio, generateImageUrl, getMaxDimensionsFromArea
 export { Link } from "./link/Link";
 export { BlockPreviewProvider } from "./preview/BlockPreviewProvider";
 export { usePreview } from "./preview/usePreview";
-export { parsePreviewParams as parsePreviewState } from "./preview/utils";
+export { parsePreviewParams as parsePreviewState } from "./preview/utils"; //legacy
```
Comment (johnnyomair) [inline packages/site/cms-site/src/index.ts:21]: Add a `@legacy` decorator here as well

File: packages/admin/cms-admin/src/preview/README.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        - sends HoverComponent on block preview hover (using HoverPreviewComponent[blocks-admin] with useIFrameBridge)
+
+```
+
+### Site: States, Contexts and Components
+
+```
+IFrameBridgePreviewPage (src/pages/preview/admin/page.tsx)
+  IFrameBridgeProvider
+      - recives messages
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/preview/README.md:58]:
```suggestion
      - receives messages
```

**Review by johnnyomair — COMMENTED** [review]
Nice refactoring! I'm not convinced whether this is not a breaking change. Could you verify this by temporary reverting your changes in Demo?

File: packages/site/cms-site/src/blocks/ExternalLinkBlock.tsx (hunk 2 above)
Comment (nsams) [reply packages/site/cms-site/src/blocks/ExternalLinkBlock.tsx:16]: fixed.

the onClick we still need - but only for preventDefault

File: packages/admin/cms-admin/src/preview/SitePreview.tsx (hunk 1 above)
Comment (nsams) [reply packages/admin/cms-admin/src/preview/SitePreview.tsx]: yes, sorry.

**Comment by nsams** [comment]
> Could you verify this by temporary reverting your changes in Demo?

is now running (I didn't test everything)

**Comment by johnnyomair** [comment]
@nsams please resolve conflicts

**Review by johnnyomair — APPROVED** [review]

**Comment by johnnyomair** [comment]
Should we add a deprecation message to the changelog?

**Comment by nsams** [comment]
> Should we add a deprecation message to the changelog?

I would not necessarily as it's fully compatible, what do you think?

**Comment by johnnyomair** [comment]
So we just add `Remove deprecated ...` when we remove it? Doing so doesn't inform users that a feature is deprecated and will be removed in the future.

I've just checked the [MUI changelog](https://github.com/mui/material-ui/blob/master/CHANGELOG.md), they do add deprecation notices

image

**Pull request merged**